optics-core/optics-core.cabal

```diff
@@ -62,12 +70,21 @@ library
                  Optics.Internal.Utils
                  Optics.Internal.View
 
-  default-extensions: FlexibleContexts
+  default-extensions: BangPatterns
+                      DefaultSignatures
+                      FlexibleContexts
+                      FlexibleInstances
+                      FunctionalDependencies
+                      DeriveFunctor
                       GADTs
+                      LambdaCase
+                      MultiParamTypeClasses
                       RankNTypes
                       ScopedTypeVariables
                       TupleSections
                       TypeFamilies
+                      TypeOperators
+                      UndecidableInstances
```

Comment:
I don't feel nice of having this in `default-extensions`.

Reply:
Personally I'm not a big fan of `default-extensions` at all, although it's less of a big deal for "simpler" extensions. At the very least, I'd agree with not enabling `UndecidableInstances` everywhere.

Reply:
> I'm fine with using UndecidableInstances on module basis, but can you elaborate why it's potentially a bad idea? After all, all it does is potentially allows instance resolution to loop.

Well, precisely because it allows instance resolution to loop. :wink: I don't think it is something to be avoided at all costs, just that one should make a conscious decision to use it for particular instances rather than having it globally enabled and hence potentially delaying finding buggy instances. Ideally GHC would even offer a per-instance pragma, but for now per-module is the closest we can get.